Sources/Instrumentation/Tracing/Span.swift

```diff
@@ -246,3 +249,24 @@ public struct SpanStatus {
         case unauthenticated
     }
 }
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: Span Kind
+
+/// Describes the relationship between the Span, its parents, and its children in a Trace.
+public enum SpanKind {
+    /// Indicates that the span covers server-side handling of a synchronous RPC or other remote request.
+    /// This span is the child of a remote `.client` span that was expected to wait for a response.
+    case server
+    /// Indicates that the span describes a synchronous request to some remote service.
+    /// This span is the parent of a remote `.server` span and waits for its response.
+    case client
+    /// Indicates that the span describes the parent of an asynchronous request. This parent span is expected to end before the corresponding child
+    /// `.consumer` span, possibly even before the child span starts. In messaging scenarios with batching,
+    /// tracing individual messages requires a new `.producer` span per message to be created.
+    case producer
+    /// Indicates that the span describes the child of an asynchronous `.producer` request.
```

Comment:
that's direct wording from OT? It's kind of bad to call those things requests to be honest... it's messages, but let's go with it ;)

Reply:
Yep, copied them over from the OT spec.